Sources/ArcGISToolkit/Components/Authentication/AuthenticationManager.swift

```
@@ -0,0 +1,82 @@
+    ///   - synchronizesWithiCloud: A Boolean value indicating whether the credentials are synchronized with iCloud.
+    func setupPersistentCredentialStorage(
+        access: ArcGIS.KeychainAccess,
+        synchronizesWithiCloud: Bool = false
+    ) async throws {
+        let previousArcGISCredentialStore = arcGISCredentialStore
+        
+        // Set a persistent ArcGIS credential store on the ArcGIS environment.
+        arcGISCredentialStore = try await .makePersistent(
+            access: access,
+            synchronizesWithiCloud: synchronizesWithiCloud
+        )
+        
+        do {
+            // Set a persistent network credential store on the ArcGIS environment.
+            await setNetworkCredentialStore(
+                try await .makePersistent(access: access, synchronizesWithiCloud: synchronizesWithiCloud)
+            )
+        } catch {
+            // If making the shared network credential store persistent fails,
+            // then restore the ArcGIS credential store.
+            arcGISCredentialStore = previousArcGISCredentialStore
+            throw error
+        }
+    }
+    
+    /// Clears all ArcGIS and network credentials from the respective stores.
+    /// Note: This sets up new `URLSessions` so that removed network credentials are respected
+    /// right away.
+    func clearCredentialStores() async {
+        // Clear ArcGIS Credentials.
+        arcGISCredentialStore.removeAll()
+        
+        // Clear network credentials.
+        await networkCredentialStore.removeAll()
+    }
+    
+    /// Revokes tokens of OAuth user credentials.
+    func revokeOAuthTokens() async {
+        let oAuthUserCredentials = arcGISCredentialStore.credentials.compactMap { $0 as? OAuthUserCredential }
+        await withTaskGroup(of: Void.self) { group in
+            for credential in oAuthUserCredentials {
+                group.addTask {
+                    try? await credential.revokeToken()
```

@rolson Ignoring the error here. However, should we go with the pattern of `Sequence.load()` and return `Bool` if one of the revoke fails?

---

yes

---

done